### PR TITLE
DPM-64 Add CLOSE_SURFACE_MODAL event, rename modal types

### PR DESCRIPTION
**Pull request by bashmach**
## Changes

- Add `CLOSE_SURFACE_MODAL` event
- Rename `Modal` type `EMBEDDED_ACTION` to `JSON_MODAL`
- Rename `Modal` type `CUSTOM_SURFACE` to `CUSTOM_MODAL`

## Related PRs
pipedrive/marketplace-app-extensions-ui#272
pipedrive/example-apps#15
pipedrive/marketplace-app-extensions-ui-blocks#97

### CODE REVIEW

**Comment by pipedrive-bot-eventsink** [comment]
### Main branch dependencies
[snyk vulnerabilities](https://app.snyk.io/org/gravity-byw/project/739623e7-a68c-463f-a3e4-a91ee075c498)

File: README.md [hunk 1]
```diff
@@ -341,3 +341,17 @@ sdk.listen(Event.VISIBILITY, ({ error, data }) => {
   // handle event
 });
 ```
+
+### Close surface modal
+
+Subscribe to own custom modals' event triggered by SDK `CLOSE_MODAL` command or user interaction with Modal.
+
+**Panel surface** - user collapses or expands the panel
```
Comment (AndreasSaltsberg) [inline README.md:349]: This line is not needed, it's only applicable for visibility event

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- New event `CLOSE_CUSTOM_MODAL` for listening the updates from custom surface modal
```
Comment (AndreasSaltsberg) [inline CHANGELOG.md:10]:
```suggestion
- New event `CLOSE_CUSTOM_MODAL` for listening the updates from custom modal
```

**Review by AndreasSaltsberg — DISMISSED** [review]
Small changes needed, in general LGTM

**Comment by pipedrive-bot-dora** [comment]
## Neodora Analysis ⚠


<details><summary><strong> ⚠ Unknown tribe name</strong></summary>
<br/>

**Message:** The property `owner` in `repository.json` must be a valid tribe name present in **mission-tracking**
**Repository file:** [repository.json](https://github.com/pipedrive/custom-app-surfaces-sdk/tree/7de33d99049ffce8b390ebfd0c5fd880553b4b98/repository.json)

<table>
<thead>
	<tr>
			<th>Owner status</th>
	</tr>
</thead>
<tbody>
		<tr>
				<td>'GRAVITY' is an unknown tribe name</td>
		</tr>
</tbody>
</table>


🔍 [How to fix it](https://neodora.pipedrive.tools/rules/tribeName)
<br/>
</details>

<details><summary><strong> ⚠ Missing cache in Github Workflow</strong></summary>
<br/>

**Message:** Github workflow should be using caching when step uses `actions/setup-node`

<table>
<thead>
	<tr>
			<th>Workflow file</th>
			<th>Job name</th>
			<th>Step index</th>
			<th>Actual cache value</th>
			<th>Expected cache value</th>
	</tr>
</thead>
<tbody>
		<tr>
				<td><a href="https://github.com/pipedrive/custom-app-surfaces-sdk/tree/7de33d99049ffce8b390ebfd0c5fd880553b4b98/.github/workflows/on-commit.yml">on-commit.yml</a></td>
				<td><code>lint</code></td>
				<td>2</td>
				<td><code>Not found</code></td>
				<td><code>npm</code></td>
		</tr>
</tbody>
</table>


🔍 [How to fix it](https://neodora.pipedrive.tools/rules/workflowCache)
<br/>
</details>


Commit [`7de33d9`](https://github.com/pipedrive/custom-app-surfaces-sdk/commit/7de33d99049ffce8b390ebfd0c5fd880553b4b98/)  Job [`28167`](https://jenkins.pipedrive.tools/job/neodora/28167/)

**Pull request merged**